src/components/shared/node_count_selector.js

```diff
@@ -28,6 +33,21 @@ class NodeCountSelector extends React.Component {
     scaling: this.props.scaling,
   };
 
+  static mergeConstraints(partialConstraints, defaultConstraints) {
```

Comment:
I feel like this is a bit more complicated than it needs to be?

Can it be just like `Object.assign({}, defaultConstraints, partialConstraints)` ?

Or am I missing a bit what this function does?

Reply:
Well it is too complicated, but it makes it easier to handle undefined input.

The actual usage example:
I only inject a `maxValue` prop in the parent component, if the value is bigger than 0 in the store. But if it's not, in reality I would just inject an object like
```
{
  max: undefined
}
```
that messes up the `NodeCountSelector` component.

The check can be done outside the `NodeCountSelector` component, and inject a `valueConstraints` prop equal to `undefined` if there is no `maxValue` in the store.
However, that check will be done multiple times, in different places. So, for less code reusability, I decided to include a safety net in the actual `NodeCountSelector` component.

So if I inject an object like 
```
{
  min: 3,
  max: undefined
}
```
The `min` value will be used, but the `max` value will be replaced by the default value.

We could handle this less painfully and have separate props for the minimul value and maximum value.

Please let me know what you think

Reply:
Separate props sounds good and would be ok with me.